pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -251,6 +254,68 @@ public CompletableFuture<Void> grantPermissionAsync(TopicName topicName, Set<Aut
         });
     }
 
+    public CompletableFuture<Void> grantPermissionAsync(List<GrantTopicPermissionOptions> options) {
+        return getPoliciesReadOnlyAsync().thenCompose(readonly -> {
+            if (readonly) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Policies are read-only. Broker cannot do read-write operations");
+                }
+                throw new IllegalStateException("policies are in readonly mode");
+            }
+            TopicName topicName = TopicName.get(options.get(0).getTopic());
+            return pulsarResources.getNamespaceResources()
+                    .setPoliciesAsync(topicName.getNamespaceObject(), policies -> {
```

Comment:
This code seems to assume that all topics are in the same namespace. That seems like a problematic assumption. Is there a goal to support `GrantTopicPermissionOptions` that apply to topics that could be in different namespaces?

Although I can see that there's a check in the different layer. The validation should happen here again unless support for multiple namespaces is added. The reason for this is that the caller shouldn't need to know implementation details. If it needs to know the implementation details, it's a leaky or broken abstraction.

Reply:
yes, Lari, should be in the same namespace.
I have checked it in the Rest Part

Reply:
https://github.com/apache/pulsar/pull/23372/files#r1794452230
@lhotari

Reply:
Thanks @lhotari , I have addressed your comment. add the validation here